dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/resource.py

```diff
@@ -28,16 +30,46 @@ class Resource(Base):
     :param name: The fullname of resource.Includes path and suffix.
     :param content: The description of resource.
     :param description: The description of resource.
+    :param user_name: The user name of resource.
     """
 
-    _DEFINE_ATTR = {"name", "content", "description"}
+    _DEFINE_ATTR = {"name", "content", "description", "user_name"}
 
     def __init__(
         self,
         name: str,
-        content: str,
+        content: Optional[str] = None,
         description: Optional[str] = None,
+        user_name: Optional[str] = None,
     ):
         super().__init__(name, description)
         self.content = content
+        self.user_name = user_name
         self._resource_code = None
+
+    def get_info_from_database(self):
+        """Get resource info from java gateway, contains resource id, name."""
+        if not self.user_name:
+            raise PyDSParamException(
+                "`user_name` is required when querying resources from python gate."
+            )
+        gateway = launch_gateway()
+        return gateway.entry_point.queryResourcesFileInfo(self.user_name, self.name)
+
+    def get_id_from_database(self):
+        """Get resource id from java gateway."""
+        return self.get_info_from_database().getId()
+
+    def create_or_update_resource(self):
+        """Create or update resource via java gateway."""
+        gateway = launch_gateway()
+        if not self.content or not self.user_name:
+            raise PyDSParamException(
+                "`user_name` and `content` are required when create or update resource from python gate."
+            )
```

Comment:
Can you add some tests when `user_name` empty, and also in function `get_info_from_database`

Reply:
Others LGTM